### ISSUE TITLE
ext_proc - sampling support

### ISSUE THREAD

**Issue by rshriram** [envoyproxy/envoy#30147]
*Title*: ext_proc - sampling support

*Description*:
We need the ability to sample ext_proc calls. A simple sample_rate field (0.0 to 1.0) within ext_proc filter that activates the filter and makes the external call only when necessary. This becomes especially useful when we finish implementing the async support, allowing users to selectively sample a fraction of traffic for observation.

Once sampled, the stream should be active for the entire request, honoring all the processing modes, etc. 

cc @tyxia @htuch

**Comment by htuch** [comment]
Yeah, preference is this is outside of ext_proc and part of the match model.

**Comment by yanjunxiang-google** [comment]
I am a little bit confused on the comments.

The original issue description is for adding sampling support for ext_proc to makes the external call only when necessary. To implement that, we can simply toss a coin in the ext_proc filter and make the call based on the coin toss result.

@kyessenov @htuch could you please elaborate the idea to implement it in the unified matcher?

**Comment by htuch** [comment]
I think @kyessenov is suggesting we just have the coin flip in the matcher, which will be more generic, as there are other legit use cases for sampling.

**Comment by tyxia** [comment]
I think it can be understood as: Perform the sampling at the higher level and centralized place. This will benefit all delegated filters (like ext_proc, ext_authz, etc)  as they are invoked based on sampling result

**Comment by yanjunxiang-google** [comment]
If we want to do sampling in xDS matcher,  a few questions are:
1) Where to store the sampling ratio config?
2) Where to execute the coin toss based on the configured sampling ratio
